cmd/demo/node.go

```diff
@@ -258,14 +262,62 @@ func (n *node) listen() error {
 	if err != nil {
 		return errors.WithMessage(err, "could not start tcp listener")
 	}
-	go n.client.HandleChannelProposals(n)
+	// Set up persistence.
+	db, err := leveldb.LoadDatabase(config.DatabasePath)
+	if err != nil {
+		return errors.WithMessage(err, "creating/loading database")
+	}
+	persister, err := keyvalue.NewPersistRestorer(db)
+	if err != nil {
+		return errors.WithMessage(err, "creating PersistRestorer")
+	}
+	n.client.EnablePersistence(persister)
+
+	n.client.OnNewChannel(func(ch *client.Channel) {
+		n.log.WithField("ID", ch.ID()).Trace("OnNewChannel")
+		go ch.Watch()
+		// Update the channel of the peer, when a peer is associated with it.
+		for _, p := range n.peers {
+			if p.ch != nil && p.ch.ID() == ch.ID() {
+				p.ch = newPaymentChannel(ch, func() {
+					n.handleFinal(p)
+				})
+				n.log.WithField("ID", ch.ID()).Debug("Channel restored")
+				return
+			}
+		}
+		// The channel came from a ProposeChannel or HandleProposal call,
+		// put it into the `pendingChannels` map.
+		n.pendingChannels[ch.ID()] = ch
```

Comment:
As discussed in chat, we should handle the channel setup directly at this point and not in `setupPendingChannel`, which is called only from `Open` and in the handler. For this, we need to add a `client.Channel.Peers()` getter. Also, right now a peer itself is not restored when a channel is restored.

Please also move this whole logic into a separate method on the node, like `node.setupChannel(*client.Channel)` that you then set as the `OnNewChannel` callback.